google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableExporterUtils.java

```diff
@@ -0,0 +1,234 @@
+  private static final long NANO_PER_SECOND = (long) 1e9;
+
+  static String getDefaultTaskValue() {
+    // Something like '<pid>@<hostname>'
+    final String jvmName = ManagementFactory.getRuntimeMXBean().getName();
+    // If not the expected format then generate a random number.
+    if (jvmName.indexOf('@') < 1) {
+      String hostname = "localhost";
+      try {
+        hostname = InetAddress.getLocalHost().getHostName();
+      } catch (UnknownHostException e) {
+        logger.log(Level.INFO, "Unable to get the hostname.", e);
+      }
+      // Generate a random number and use the same format "random_number@hostname".
+      return "java-" + new SecureRandom().nextInt() + "@" + hostname;
+    }
+    return "java-" + UUID.randomUUID() + jvmName;
+  }
+
+  private static final Set<AttributeKey<String>> PROMOTED_RESOURCE_LABELS =
+      ImmutableSet.of(PROJECT_ID, INSTANCE_ID, TABLE_ID, CLUSTER_ID, ZONE_ID);
+
+  static TimeSeries convertPointToTimeSeries(
+      MetricData metricData,
+      PointData pointData,
+      String taskId,
+      MonitoredResource monitoredResource) {
+    Map<AttributeKey<?>, Object> attributes = pointData.getAttributes().asMap();
+    MonitoredResource.Builder monitoredResourceBuilder = monitoredResource.toBuilder();
+    ImmutableMap.Builder<String, String> metricLabels = new ImmutableMap.Builder<>();
+    for (AttributeKey<?> attributeKey : attributes.keySet()) {
+      if (PROMOTED_RESOURCE_LABELS.contains(attributeKey)) {
+        monitoredResourceBuilder.putLabels(
+            attributeKey.getKey(), String.valueOf(attributes.get(attributeKey)));
+      } else {
+        if (attributes.get(attributeKey) != null) {
+          metricLabels.put(attributeKey.getKey(), String.valueOf(attributes.get(attributeKey)));
+        }
+      }
+    }
+    metricLabels.put(CLIENT_UID.getKey(), taskId);
+
+    TimeSeries.Builder builder =
+        TimeSeries.newBuilder()
+            .setResource(monitoredResourceBuilder.build())
+            .setMetricKind(convertMetricKind(metricData))
+            .setMetric(
+                Metric.newBuilder()
+                    .setType(metricData.getName())
+                    .putAllLabels(metricLabels.build())
+                    .build())
+            .setValueType(convertValueType(metricData.getType()));
+
+    TimeInterval timeInterval =
+        TimeInterval.newBuilder()
+            .setStartTime(convertTimestamp(pointData.getStartEpochNanos()))
+            .setEndTime(convertTimestamp(pointData.getEpochNanos()))
+            .build();
+
+    builder.addPoints(createPoint(metricData.getType(), pointData, timeInterval));
+
+    return builder.build();
+  }
+
+  static String getProjectId(PointData pointData) {
+    return pointData.getAttributes().get(PROJECT_ID);
+  }
+
+  private static MetricKind convertMetricKind(MetricData metricData) {
+    switch (metricData.getType()) {
```

Comment:
It seems like all of our supported metrics CUMULATIVE AggregationTemporality. Why not make it the first check in this function?

Reply:
AggregationTemporality is not an option for all data types, so I need to first get the data type and then check the AggregationTemporality :(